src/components/navigation-bar/types.ts

```diff
@@ -0,0 +1,21 @@
+import { ReactNode } from "react";
+
+export type NavigationItem<T = unknown> = T & {
+  id: string;
+  label: ReactNode;
+  isSelected?: boolean;
+  disabled?: boolean;
+  children?: Array<NavigationItem<T>>;
+};
+
+export type NavigationBarProps = {
+  items: NavigationItem[];
+  onItemClick?: (item: NavigationItem) => void;
+  onLogin?: () => void;
+  isAuth?: boolean;
+  authDropdownContainer?: ReactNode;
+  brand?: ReactNode;
+  username?: string;
+  noLogin?: boolean;
+  className?: string;
```

Comment:
Shouldn't we restrict ```className``` mandatory? Because we should customize components only by using overrides.
@KlonD90

Reply:
In other case there would be a crazy mix of customizing approaches inside one project

Reply:
I don't think that we need to restrict it like everywhere. ClassName is still applicable as a thing. And could be use in a lot of things around not only customising but also writing tests etc etc. Like we don't want to introduce overrides to someone who just want to write some tests for behaviour

Reply:
About testing - in that case it's more convenient to allow all `data-attributes` in the props, because another common approach is to use them.

Reply:
In general it's not a big issue to have `className`, we could fix that in next releases, this should not block component to merge, so i'm closing that comment, may return later.